lightning/config/config.go

```diff
@@ -204,8 +205,58 @@ func (cfg *Config) LoadFromGlobal(global *GlobalConfig) error {
 }
 
 // LoadFromTOML overwrites the current configuration by the TOML data
+// If data contains toml items not in Config and GlobalConfig, return an error
+// If data contains toml items not in Config, thus won't take effect, warn user
 func (cfg *Config) LoadFromTOML(data []byte) error {
-	return errors.Trace(toml.Unmarshal(data, cfg))
+	// warn legal toml items but won't effect
+	var warnItems []string
+	dataStr := string(data)
+	metaData, err := toml.Decode(dataStr, cfg)
+
+	// Here's a warning that we haven't check err before use metaData,
+	// but metaData will always return a valid struct and we check err in next line
+	undecoded := metaData.Undecoded()
```

Comment:
the variable name seems not good enough

Reply:
how about change `undecoded` to `unused`